core/templates/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -78,7 +79,7 @@ export interface InteractionBackendDict {
   'confirmed_unclassified_answers': readonly InteractionAnswer[];
   'customization_args': InteractionCustomizationArgsBackendDict;
   'hints': readonly HintBackendDict[];
-  'id': string;
+  'id': keyof typeof InteractionSpecsConstants.INTERACTION_SPECS;
```

Comment:
Should we add the `InteractionId` above and then use it everywhere?
```typescript
export type InteractionId = (
  keyof typeof InteractionSpecsConstants.INTERACTION_SPECS)
```

Reply:
yes, that would be better. I've added the changes